Publish Finance Documents to the Site

saranshgrover opened the pull request:
Closes #6038 

Since all the finance documents are PDFs, I have added them to this repository in the public/files/finances folder. In order to get the `document_list` method to work with this, I have changed the path to explicitly describe the entire path (previously it pre-filled into the documents path). I have changed all function calls to reflect this.

Note that "Finances" isn't i18n'd so I just entered the english word in this case - is there a proper way to handle it?

viroulep commented:
Is there any issue with "just" creating a "finances" folder within the "documents" folder of WCA-documents?
If I'm not mistaken the build script there just deploy the pdf as-is if there are any!

The diff here would be minimal (just adding the extra finance section with listing), and most importantly there are two advantages:
  - all the official documents are gathered at the same place
  - you don't need a subsequent PR here if you want to add/remove/change finances documents, you just have to make the changes there and they will be deployed.

> Note that "Finances" isn't i18n'd so I just entered the english word in this case - is there a proper way to handle it?

Absolutely! You can add an entry in the `documents` section [here](https://github.com/thewca/worldcubeassociation.org/blob/7fe3bbd365f93c38cc1e33a32d4a9887f7265bbe/WcaOnRails/config/locales/en.yml#L1941), and use `t('documents.finance')` to fetch it (similar to what has been done for the other sections).
Translators are informed when there are changes to this file, so that they can provide the translation whenever they can.

saranshgrover commented:
Thanks - I missed that you could simply put pdfs in wca-documents. I've removed the files from here and added them in thewca/wca-documents#236 and also updated `en.yml`

viroulep commented on WcaOnRails/app/helpers/documents_helper.rb:
```diff
@@ -2,9 +2,9 @@
 module DocumentsHelper
   def documents_list(directory)
-    safe_join Dir.glob("#{Rails.root}/public/documents/#{directory}/*.pdf")
+    safe_join Dir.glob("#{Rails.root}/public/#{directory}/*.pdf")
```
Given everything is still nested within `documents` I believe this is not necessary anymore :)

saranshgrover replied:
Reverted

viroulep reviewed (commented):
Thanks for the changes, looks way cleaner to me!
If you don't mind I think you can revert the changes where you browse all the public folder, since we always give a path starting by `documents` to that method; and then we can squash everything to have just a single clean commit.

saranshgrover commented:
This should be ready to be merged @thewca/software-team whenever you get a chance!

denthebro commented:
@saranshgrover this did indeed cause an issue. The way the builder works right now is it creates temporary PDFs in the documents folder, copies them into build and then deletes all PDFs in documents. I have changed the builder [here](https://github.com/thewca/wca-documents/pull/244) and moved the finance documents to a new "static" folder. All pre-rendered PDFs should be stored here; the builder copies them at the end, so they will be found in build/static/finances.

denthebro commented:
I have taken a look at your code and the link wouldn't work if the builder update was merged. I will change the builder to make it copy folders inside of the static folder into the documents folder so that you don't have to update your code in any way.

denthebro commented:
I don't think it would work without my changes btw, because the finances folder gets deleted by the builder in its current state.

saranshgrover commented:
Got it - let me know if I need to change to use the static path instead.

denthebro commented:
@saranshgrover no, it should work with my new commit to the PR I linked. I just think you should wait with merging this PR until my PR is merged, because your PDF files will just be deleted by the builder in its current state.

denthebro commented:
Sorry, I just checked the build branch and it looks like they magically got added. I'm not really sure why it works in the repo, but regardless, my PR being merged won't affect your link to these documents.

denthebro commented:
Also, I looked at your code and it looks good to me. I'm not a WST member though, so I don't know if that means anything, haha.

viroulep commented:
I'm not sure I understand the discussion which is going on here, but I'm positive both pull requests from @saranshgrover (https://github.com/thewca/wca-documents/pull/236, and this one) were absolutely fine and working well!

Having pdf in the existing source tree works because when creating the `build` folder we copy everything from the original `documents` folder [here](https://github.com/thewca/wca-documents/blob/89efe7658c5d6b71f16342a513f5c1499f83193e/bin/build.sh#L48).
The process is as follow:
  - convert all md file in the source tree to pdfs
  - copy the whole source tree to a new build folder
  - cleanup the build folder by removing non-pdf file from there
  - cleanup the source folder by removing build files (html/pdf)

Writing this I realize that we should probably copy the whole source tree to the build folder first, then build and cleanup within the build folder.
That way we would avoid touching anything at all in the original source folder; I'll make some comments over in https://github.com/thewca/wca-documents/pull/244/files.

This PR definitely can (and should) be squashed-merged in the current state!

gregorbg commented:
Agreed about (squash-)merging this. Copying everything into the `build` folder _before_ triggering the Markdown compilation chain is a brilliant idea that none of us had thought of before. But let's continue the discussion over at the other issue!
